src/mock.js

```diff
@@ -82,6 +82,21 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      const oldResolveType = namedFieldType.resolveType;
+      // the default `resolveType` always returns null. We add a fallback
+      // resolution that works with how unions and interface are mocked
+      namedFieldType.resolveType = (data, context, info) =>
+        oldResolveType(data, context, info) || info.schema.getType(data.typename);
```

Comment:
Sorry, I just realized that what I said before doesn't make sense. If `preserveResolvers` is true, then this code works, but if `preserveResolvers` is false, we should always overwrite the existing `resolveType`, otherwise it might not work, because the resolver would be changed, but the `resolveType` function would still be the old one. Can you change it to just return `info.schema.getType(data.typename)`, but wrap the function call in an if statement? (I made another comment below). Thanks!